### DISCUSSION TITLE
DRF support

### DISCUSSION THREAD

**Discussion by justquick** [justquick/django-activity-stream#514] [Ideas]
django rest framework support for Follow/Action in actstream mod
maybe try token auth in runtests

**Comment by rollue** [comment]
This looks like a promising starting point.
[Django Activity Stream with Django REST Framework](https://andrew.hawker.io/writings/2020/11/16/django-activity-stream-with-django-rest-framework/)

**Comment by justquick** [comment]
yes ive reached out to @ahawker via email about collaborating but havent gotten a response. failing that i will take what he started an adapt it into the project. a first round prototype has been pretty promising but has some more things to work out

**Comment by auvipy** [comment]
just start with draft, we can improve it over time. https://github.com/Ian-Foote/rest-framework-generic-relations & https://www.django-rest-framework.org/api-guide/relations/#generic-relationships are helpful refs.

**Comment by justquick** [comment]
I have fallen in love with this project and am working to annotate the app with new types/schemas/etc. Its going to have some lock in like the others but it provides rich support for OpenAPI3 and client generators 

https://github.com/tfranzel/drf-spectacular
